src/stdune/path.ml

```diff
@@ -366,6 +441,10 @@ end = struct
     | _, true -> a
     | _, _ -> make ((to_string a) ^ "/" ^ (to_string b))
 
+  let append_relative a b =
```

Comment:
`append_relative` seems misleading. It's easy to expect `append_relative "a/b" ".." = "a"`, but it actually raises, doesn't it?